operator/internal/manifests/storage/schema.go

```diff
@@ -0,0 +1,200 @@
+package storage
+
+import (
+	"sort"
+	"time"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	lokiv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+)
+
+const (
+	// DateTimeFormat is the datetime string need to format the time.
+	DateTimeFormat = "2006-01-02"
+
+	// UpdateDelay is the time before a change to the storage schema
+	// is applied to a cluster. This will avoid scenarios in which
+	// the difference between UTC and local time zones will cause
+	// logs to be unavailable.
+	UpdateDelay = time.Hour * 24
+)
+
+type schemaConfig struct {
+	version       lokiv1beta1.ObjectStorageSchemaVersion
+	effectiveDate time.Time
+}
+
+// BuildSchemaConfigList creates a list of schemas to be used to configure
+// the storage schemas for the cluster. This method assumes that the following
+// validation has been done to the statuses and specs:
+//
+// 1. All EffectiveDate fields are able to be parsed
+// 2. All EffectiveDate fields are unique in their respective list
+//
+func BuildSchemaConfigList(
```

Comment:
nit. `BuildSchemaConfig` is imho sufficient as we re-use the loki configuration terminology. `List` as a suffix is not adding more information to the calling side to execute as per return value already a slice. Suffixes like `List` for functions/methods are usually used in dynamic typed programming languages. Here is Go-Land and we say what we do with static types :)